src/urllib3/util/util.py

```diff
@@ -0,0 +1,36 @@
+def ensure_binary(s, encoding="utf-8", errors="strict"):
```

Comment:
Thoughts on using `to_str` and `to_bytes` as both shorter and more descriptive?

The following are optimized such that `to_X()` on type `X` is fast and we only pay the "not isinstance()" tax when it's not `X`. Also calling without any arguments is a lot faster than with any arguments so we should optimize for this case:

```python
from typing import Union, Optional


def to_bytes(x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None) -> bytes:
    if isinstance(x, bytes):
        return x
    elif not isinstance(x, str):
        raise TypeError(f"not expecting type {type(x).__name__}")
    elif encoding or errors:
        return x.encode(encoding, errors=errors)
    return x.encode()


def to_str(x: Union[str, bytes], encoding: Optional[str] = None, errors: Optional[str] = None) -> str:
    if isinstance(x, str):
        return x
    elif not isinstance(x, bytes):
        raise TypeError(f"not expecting type {type(x).__name__}")
    elif encoding or errors:
        return x.decode(encoding, errors=errors)
    return x.decode()
```

That way `to_bytes()` will encode as ASCII with errors=strict

Reply:
@sethmlarson Thanks for the suggestion.
I would suggest using default values for `encoding` and `errors` because with 
```python
    elif encoding or errors:
        return x.decode(encoding, errors=errors)
````
If we pass just `encoding`, the condition will be True and we will face `TypeError` (because the `error` is None)
`TypeError: decode() argument 2 must be str, not None`

Reply:
Ah yes, perhaps `x.encode(encoding or "ascii", errors=errors or "strict")`. Trying to detect the case when we can simple `x.decode()` since it's >4x faster than calling with any arguments and encoding/decoding is a common operation for us.